model/riscv_insts_zvkned.sail

```diff
@@ -0,0 +1,428 @@
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let LMUL     = if LMUL_pow < 0 then 0 else LMUL_pow;
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  if (zvk_check_elements(VLEN, num_elem, LMUL, SEW) == false)
+  then {
+    handle_illegal();
+    RETIRE_FAIL
+  } else {
+    let 'n = num_elem;
+    let 'm = SEW;
+    assert('m == 32);
+
+    let vs2_val : vector('n, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+    let vd_val  : vector('n, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+
+    let eg_len = (unsigned(vl) / 'n);
+    let eg_start = (unsigned(vstart) / 'n);
+    let keyelem = if suffix == "vv" then 1 else 0;
+
+    foreach (i from eg_start to (eg_len - 1)) {
+      assert(0 <= ((i * 4) + 3) & ((i * 4) + 3) < 'n);
+
+      let vd_state = get_velem(vd_val, i);
+      let vs2_key = get_velem(vs2_val, keyelem * i);
+      let sb       : bitvector(128) = aes_subbytes_fwd(vd_state);
+      let sr       : bitvector(128) = aes_shift_rows_fwd(sb);
+      let ark      : bitvector(128) = sr ^ vs2_key;
+
+      write_single_element(128, i, vd, ark);
+    };
+
+    set_vstart(zeros());
+    RETIRE_SUCCESS
+  }
+}
+
+/* VAESEM.[VV, VS] */
+
+mapping vaesem_mnemonic : bits(7) <-> string = {
+  0b1010001 <-> "vaesem.vv",
+  0b1010011 <-> "vaesem.vs",
+}
+
+union clause ast = RISCV_VAESEM : (regidx, regidx, string)
+
+mapping clause encdec = RISCV_VAESEM(vs2, vd, suffix)          if extensionEnabled(Ext_Zvkned)
+ <-> vv_or_vs(suffix) @ vs2 @ 0b00010 @ 0b010 @ vd @ 0b1110111 if extensionEnabled(Ext_Zvkned)
+
+mapping clause assembly = RISCV_VAESEM(vs2, vd, suffix)
+ <-> vaesem_mnemonic(vv_or_vs(suffix)) ^ spc() ^ vreg_name(vd)
+                                       ^ sep() ^ vreg_name(vs2)
+
+function clause execute (RISCV_VAESEM(vs2, vd, suffix)) = {
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let LMUL     = if LMUL_pow < 0 then 0 else LMUL_pow;
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  if (zvk_check_elements(VLEN, num_elem, LMUL, SEW) == false)
+  then {
+    handle_illegal();
+    RETIRE_FAIL
+  } else {
+    let 'n = num_elem;
+    let 'm = SEW;
+    assert('m == 32);
+
+    let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+    let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+
+    let eg_len = (unsigned(vl) / 'n);
+    let eg_start = (unsigned(vstart) / 'n);
+    let keyelem = if suffix == "vv" then 1 else 0;
+
+    foreach (i from eg_start to (eg_len - 1)) {
+      assert(0 <= ((i * 4) + 3) & ((i * 4) + 3) < 'n);
+
+      let vd_state = get_velem(vd_val, i);
+      let vs2_key  = get_velem(vs2_val, keyelem * i);
+      let sb       : bitvector(128, dec) = aes_subbytes_fwd(vd_state);
+      let sr       : bitvector(128, dec) = aes_shift_rows_fwd(sb);
+      let mix      : bitvector(128, dec) = aes_mixcolumns_fwd(sr);
+      let ark      : bitvector(128, dec) = mix ^ vs2_key;
```

Comment:
`bits(128)`. You could just delete all these, though it may be appropriate for readability.